pallets/subtensor/tests/sudo.rs

```diff
@@ -617,4 +617,124 @@ pub fn test_sudo_test_pending_emission_ok() {
         assert_eq!(SubtensorModule::get_pending_emission(netuid1), 0); // emission drained at block 3 for tempo 5
         assert_eq!(SubtensorModule::get_pending_emission(netuid2), 2250000000); // 750000000 + 750000000 + 750000000
     });
+}
+
+#[test]
+// Required by the test otherwise it would panic if compiled in debug mode
+#[allow(arithmetic_overflow)]
+fn test_set_emission_values_errors_on_emission_sum_overflow() {
+    new_test_ext().execute_with(|| {
+        let netuids: Vec<u16> = vec![ 1,2 ];
+        // u64(u64::MAX + 1..000..1) equals to 1_000_000_000 which is the same as
+        // the value of Self::get_block_emission() expected by the extrinsic
+        let emission: Vec<u64> = vec![ u64::MAX, 1_000_000_001 ];
+        add_network(1, 0, 0);
+        add_network(2, 0, 0);
+        assert_eq!(
+            SubtensorModule::sudo_set_emission_values(<<Test as
+Config>::RuntimeOrigin>::root(), netuids, emission ),
+            Err(Error::<Test>::InvalidEmissionValues.into())
+        );
+ });
+}
+
+#[test]
+#[allow(arithmetic_overflow)]
+fn test_set_emission_values_no_errors() {
+    new_test_ext().execute_with(|| {
+        let netuids: Vec<u16> = vec![ 1,2 ];
+        let emission: Vec<u64> = vec![ 600_000_000, 400_000_000 ];
+        
+        add_network(1, 0, 0);
+        add_network(2, 0, 0);
+        assert_eq!(
+            SubtensorModule::sudo_set_emission_values(<<Test as
+Config>::RuntimeOrigin>::root(), netuids, emission ),
+            Ok(())
+        );
+ });
+}
```

Comment:
we have similar tests for setting emissions in tests/networks.rs. could be more organized to keep tests related to the networks in networks test file. not a big deal though :)

Reply:
I moved them in the next commit: https://github.com/opentensor/subtensor/pull/116/commits/c04f806994e2c6c4826993fccd3b3f3e930b6aa5